Orange/widgets/visualize/owlinearprojection.py

```diff
@@ -1,181 +1,252 @@
+        n_all_attrs = len(self.attrs)
+        n_attrs = self.n_attrs
+        return ceil(0.5 * factorial(n_all_attrs) / (factorial(n_all_attrs - n_attrs) * n_attrs))
+
+    def iterate_states(self, state):
+        if state is None:  # on the first call, compute order
+            attrs = self._score_heuristic()
+            self.attrs = attrs
+        for c in combinations(attrs, self.n_attrs):
+            for p in islice(permutations(c[1:]), ceil(factorial(len(c) - 1)/2)):
+                yield (c[0], ) + p
+
+    def compute_score(self, state):
+        master = self.master
+        variables = [s for s in state]
+        master.prepare_plot_data(variables)
+        X = master.plotdata.embedding_coords
+        Y = master.data.Y
+        if X.shape[0] < self.minK:
+            return
+        n_neighbors = min(self.minK, len(X) - 1)
+        knn = NearestNeighbors(n_neighbors=n_neighbors).fit(X)
+        ind = knn.kneighbors(return_distance=False)
+        if master.data.domain.has_discrete_class:
+            return -np.sum(Y[ind] == Y.reshape(-1, 1)) / n_neighbors / len(Y)
+        else:
+            return -r2_score(Y, np.mean(Y[ind], axis=1)) * \
+                   (len(Y) / len(master.data))
+
+    def _score_heuristic(self):
+        def normalized(col):
+            if not col.size:
+                return col.copy()
+            col_min, col_max = np.nanmin(col), np.nanmax(col)
+            if np.isnan(col_min):
+                return col.copy()
+            span = col_max - col_min
+            return (col - col_min) / (span or 1)
+        mdomain = self.master.data.domain
+        keys = [i for i, a in enumerate(mdomain.attributes) if a.is_continuous]
+        domain = Domain([mdomain[i] for i in keys], mdomain.class_var)
+        data = self.master.data.transform(domain)
+        for i, col in enumerate(data.X.T):
+            data.X.T[i] = normalized(col)
+        relief = ReliefF if mdomain.class_var.is_discrete else RReliefF
+        weights = relief(n_iterations=100, k_nearest=self.minK)(data)
+        attrs = sorted(zip(weights, mdomain.attributes),
+                       key=lambda x: (-x[0], x[1].name))
+        return [a for _, a in attrs]
+
+    def row_for_state(self, score, state):
+        variables = [s for s in state]
+        item = QStandardItem(", ".join(s.name for s in state))
+        item.setData(variables, self._AttrRole)
+        return [item]
+
+    def on_selection_changed(self, selected, deselected):
+        attrs = selected.indexes()[0].data(self._AttrRole)
+        self.selectionChanged.emit([attrs])
+
+    def _n_attrs_changed(self):
+        if self.n_attrs != self.last_run_n_attrs or self.saved_state is None:
+            self.button.setText("Start")
+        else:
+            self.button.setText("Continue")
+        self.button.setEnabled(self.check_preconditions())
+
+
+class LinProjInteractiveViewBox(InteractiveViewBox):
+    def _dragtip_pos(self):
+        return 10, 10
```

Comment:
This is for dragging axes? Does linear projection allow it? (Currently not, I guess.)